proposals/0108-transaction-size-fee.md

```diff
@@ -0,0 +1,190 @@
+[this comment](https://github.com/solana-labs/solana/issues/28063#issuecomment-1727151064):
+
+Without congestion control, there is slight decrease in transactions per second
+with more padding bytes. However, the overall variance is larger than any effect
+from padding bytes. So there may be some negligible effect based on transaction
+size.
+
+This test, however, was only on a small network, so we can't come to any
+conclusions about how larger transactions will impact a real network.
+
+## New Terminology
+
+Transaction size fee: the additional lamports paid by a user, based on the number
+of bytes in the serialized transaction.
+
+## Detailed Design
+
+### Fee Calculation
+
+We propose a solution with a few important features:
+
+* the new fee does not impact current transactions
+* the fee contains a non-linear component to dissuade huge transactions,
+especially while we see the impact of large transactions on a real cluster
+* the fee also includes a linear component for the future, if the effect on a
+real cluster is small
+
+Given `total_transaction_bytes` and the fee parameters
+`square_lamports_per_byte` and `linear_lamports_per_byte`, the fee is given as:
+
+```
+transction_bytes_over_mtu = total_transaction_bytes - 1232
+linear_fee = ceil(linear_lamports_per_byte * transaction_bytes_over_mtu)
+square_fee = ceil((square_lamports_per_byte * transaction_bytes_over_mtu) ^ 2)
+transaction_size_fee = linear_fee + square_fee
+```
+
+The `linear_lamports_per_byte` and `square_lamports_per_byte` can be modified by
+the cluster, similar to `lamports_per_signature`.
+
+In this proposal, we err on the side of caution in order to make larger
+transactions quadratically more expensive, and omit the linear component.
+
+* `linear_lamports_per_byte`: 0
+* `square_lamports_per_byte`: 0.05
+
+This gives the following fees for different transaction sizes, in lamports, and
+a comparison to the current base transaction fee of 5,000 lamports:
+
+Tx Multiple | Tx Byte Size | Size Fee (Lamports) | ~Fee / 5,000
+--- | --- | --- | ---
+2x | 2,464 | 3,795 | 0.75x
+3x | 3,696 | 15,179 | 3x
+5x | 6,160 | 60,713 | 12x
+10x | 12,320 | 307,360 | 62x
+20x | 24,640 | 1,369,837 | 274x
+
+Again, the approach of a non-linearly increasing fee is very conservative, to
+properly compensate leaders if large transactions reduce cluster TPS or
+significantly increase bandwidth usage.
+
+Over time, as the impact of large transactions on a cluster is measured
+concretely, the fee could be greatly reduced, or switched to the linear
+coefficient only.
+
+The fee coefficients can be updated through a stake-weighted vote from the
+validator set, but the update process is out of scope for this document.
+
+### Validator Changes
+
+When calculating the fee, the transaction size fee must be calculated and deducted
+from the fee payer account as outlined earlier.
+
+As with other fees, half of the fee is burned, and half goes to the leader.
```

Comment:
this breaks prioritization.  should leaders do priority * CUs + 1/2 * base + 1/2 * tx size / CUs as the priority function?

the fee doesn't make sense and is out of the blue tbh.  it should just be a CU modifier based on benchmarks.